goaviatrix/firewall_instance.go

```diff
@@ -180,6 +197,13 @@ func (c *Client) GetFirewallInstance(firewallInstance *FirewallInstance) (*Firew
 		return nil, errors.New("Rest API get_instance_by_id Get failed: " + data.Reason)
 	}
 	if data.Results.InstanceID == firewallInstance.InstanceID {
+		// Only try to decode if tags are not empty string
+		if string(data.Results.TagsMessage) != `""` {
+			err = json.Unmarshal(data.Results.TagsMessage, &data.Results.Tags)
+			if err != nil {
+				return nil, fmt.Errorf("json Decode get_instance_by_id failed: %v", err)
+			}
+		}
```

Comment:
I submitted a mantis for this: 20558. I think the backend should return an empty dict instead of an empty string so we don't have this problem.

Reply:
I agree. I will keep this workaround for now until 20558 is fixed.